Exclude database cache file from iCloud backup

namannik opened the pull request:
My app was initially rejected because the `RMDatabaseCache` file was being backed up to iCloud.  Adding a line of code to exclude the file from backups resolved the problem.

Commit:
Exclude database cache file from iCloud backup

incanus commented:
Thank you for this, but there is backstory at https://github.com/mapbox/mapbox-ios-sdk/issues/354 which is partially lost due to a support system migration. In short, we can't just make this change across the board but need to provide configurability to the developer depending upon their situation: 
1. They treat tiles simply as cache and **do not** wish them to be backed up. 
2. They treat tiles as documents that make key functionality available when offline and **do** wish them to be backed up. 

Would you be interested in augmenting your patch to provide that sort of configurability?

namannik commented:
Yeah, I can take a look at doing that.  Which case do you feel should be the default?  My opinion, of course, would be for the default to not backup.

incanus commented:
Hmm, I would agree with you then, as I've only heard tell of review problems, not data loss. We'll just have to make sure to document it well in the offline caching-related docs.

namannik commented:
Looking at `RMDatabaseCache`, I see that there's an initialization method called `initUsingCacheDir:useCacheDir`, which I've never paid attention to before.  Data stored in this directory is not included in iCloud backups.

If a new option is added to exclude cache stored in the _documents_ directory from backup, is this redundant with `initUsingCacheDir:useCacheDir`?  Is there a use case where the cache should not be stored in the cache directory and also not backed up to iCloud?

In my case, I'd be fine storing the cache in the cache directory, but `RMMapView` doesn't currently provide a method to initialize with this option.

Commit:
Revert "Exclude database cache file from iCloud backup"

This reverts commit 4b0d15fb96dec26dcd0eef6123cec6198d46fbe4.

Commit:
Exclude db cache from iCloud backups with option to include

When a new RMTileCache is created, the db cache will be excluded from
automatic iCloud backups by default.  To include it in backups, call
includeInBackup:.

namannik commented:
Closing in order to create new pull request per suggestions above by @incanus.
